packages/mui-material/src/styles/createPalette.js

```diff
@@ -256,7 +264,7 @@ export default function createPalette(palette) {
     return color;
   };
 
-  const modes = { dark, light };
+  const modes = { dark: getDark(), light: getLight() };
```

Comment:
We should refactor this then? It sounds wasteful to call `getDark()` to then throw away its value.